### PR TITLE
jz - added component to submit the MilktheCowsJob

### PR DESCRIPTION
**Pull request by xxzjh**
Screen Shot 2022-11-16 at 7 00 18 PM

replace the origin placeholder <JobComingSoon /> into the new placeholder to launch the "Milk the Cows" job

### CODE REVIEW

**Review by pconrad — CHANGES_REQUESTED** [review]
Could we rename "JobMilktheCows.js to MilkCowsJobForm.js so that it is consistent with TestJobForm.js?

Also, could we go ahead and put this in the component?  That way the component is ready to go when we are ready to hook it up to the backend.

```jsx
 
 <Form onSubmit={handleSubmit(submitAction)}>
      <Button type="submit" data-testid="MilkCowsForm-Submit-Button">Submit</Button>
 </Form>

```

File: frontend/src/main/pages/AdminJobsPage.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { useBackend } from "main/utils/useBackend";
 import Accordion from 'react-bootstrap/Accordion';
 import TestJobForm from "main/components/Jobs/TestJobForm";
 import JobComingSoon from "main/components/Jobs/JobComingSoon";
+import JobMilktheCows from "main/components/Jobs/JobMilktheCows";
 
 import { useBackendMutation } from "main/utils/useBackend";
```
Comment (pconrad) [inline frontend/src/main/pages/AdminJobsPage.js:11]: Let's also add this---it's a copy/paste/edit of the code for the test job:

```jsx
 // test job 

    const objectToAxiosParamsMilkCowsJob = (data) => ({
        url: `/api/jobs/launch/milkcows`,
        method: "POST"
    });

    // Stryker disable all
    const milkCowsJobMutation = useBackendMutation(
        objectToAxiosParamsMilkCowsJob,
        {  },
        ["/api/jobs/all"]
    );
    // Stryker enable all

    const submitMilkCowsJob = async (data) => {
        console.log("submitMilkCowsJob, data=", data);
        milkCowsJobMutation.mutate(data);
    }

```

Then, change this line: 

```jsx
form: <JobMilktheCows />
```

To this one: 

```jsx
form: <JobMilktheCows submitAction={submitMilkCowsJob}
```

That way, the entire frontend is ready to go... as soon as the backend is complete, the job will work.

**Review by pconrad — CHANGES_REQUESTED** [review]
There are a few change that will make this PR a bit more complete.

File: frontend/src/main/components/Jobs/MilkCowsJobForm.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Button, Form } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+
+function MilkCowsJobForm(submitAction) {
+  const defaultValues = {
+    fail: false,
+    sleepMs: 1000
+};
```
Comment (pconrad) [inline frontend/src/main/components/Jobs/MilkCowsJobForm.js:8]: We don't need these values for milking the cows; these are only needed for the test job.   We can remove these.

File: frontend/src/main/components/Jobs/MilkCowsJobForm.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  // Stryker disable all
+  const {
+    handleSubmit,
+  } = useForm(
+    { defaultValues: defaultValues }
```
Comment (pconrad) [inline frontend/src/main/components/Jobs/MilkCowsJobForm.js:13]: For now, this can just be: 
```js
 {defaultValues: {} }
```

**Review by pconrad — CHANGES_REQUESTED** [review]

**Review by pconrad — APPROVED** [review]
LGTM

**Review by BenYangSB — COMMENTED** [review]
looks good

**Pull request merged**